pkg/proxy/endpointslicecache.go

```diff
@@ -249,5 +249,5 @@ func (e byIP) Swap(i, j int) {
 	e[i], e[j] = e[j], e[i]
 }
 func (e byIP) Less(i, j int) bool {
-	return e[i].IP() < e[j].IP()
+	return e[i].String() < e[j].String()
```

Comment:
We could add `CompareByIP(Endpoint* other) int` to the Endpoint interface and implement with bytes.compare() to avoid allocating a string on each comparison.

Reply:
@bowei This actually is just returning an IP:Port string that happens to be the only string stored with this endpoint struct. Previously when using the `.IP()` method it was actually resulting in a netutil call to parse that out which was resulting in the slowness. Here it's just returning the string that is already stored as is.

Reply:
Ah right, I thought for some reason it was stored as an IP internally.